src/api/controllers/IdentityController.js

```diff
@@ -118,24 +177,63 @@ class IdentityController extends PointSDKController {
     }
 
     async isIdentityRegistered() {
-        const identityRegistred = await blockchain.isCurrentIdentityRegistered();
-        return this._response({identityRegistred: identityRegistred});
+        const identity = await getIdentity();
+        return this._response({
+            identityRegistred: Boolean(identity),
+            identity
+        });
     }
 
     async identityToOwner() {
         const identity = this.req.params.identity;
-        const owner = await blockchain.ownerByIdentity(identity);
-        return this._response({owner: owner});
+
+        let owner = '';
+        let network = '';
+        if (identity.endsWith('.sol')) {
+            const registry = await solana.resolveDomain(identity);
+            owner = registry.owner;
+            network = 'solana';
+        } else if (identity.endsWith('.eth')) {
+            const registry = await blockchain.resolveDomain(identity);
+            owner = registry.owner;
+            network = 'ethereum';
+        } else {
+            owner = await blockchain.ownerByIdentity(identity);
+            network = 'point';
+        }
+
+        return this._response({owner, network});
     }
 
     async ownerToIdentity() {
         const owner = this.req.params.owner;
-        const identity = await blockchain.identityByOwner(owner);
+
+        let key = '';
+        let targetOnly = '';
+        if (solana.isAddress(owner)) {
+            key = 'solAddress';
+            targetOnly = 'solana';
+        } else if (blockchain.isAddress(owner)) {
+            key = 'ethAddress';
+            targetOnly = 'ethereum';
+        }
+
+        if (!key || !targetOnly) {
+            this.rep.status(400);
+            return this._status(400)._response('Invalid owner address provided.');
+        }
+
+        const identity = await getIdentity({[key]: owner, targetOnly});
         return this._response({identity});
     }
 
     async publicKeyByIdentity() {
         const identity = this.req.params.identity;
+        if (identity.endsWith('.sol') || identity.endsWith('.eth')) {
+            this.rep.status(422);
+            return this._status(422)._response('This endpoint only supports Point identities.');
```

Comment:
For Ethereum, I don't think it's possible to get a Public Key from an Address. Addresses are derived from Public Keys but have "less data", so going the other way is not possible. I assume that's the reason we keep the Public Keys in the Identity Contract.

For Solana, I understand the Public Key and the Address are the same thing, so we shouldn't have this problem. I would need to confirm it though.

When the identity is an ENS domain, this is going to be a problem for dApps that need the Public Key. For example _email.point_ and most likely _drive.point_.

If they only need the public key of the "logged in" user, we can get it because we have the secret phrase. But, if they need one from another ENS identity, to encrypt an email (or a shared file) for example, we won't be able to get it this way.

For SNS identities, we need to check if encrypting with Solana Public Keys works as expected.

Reply:
Yes, I think this method should only be used for Point identities. Let's throw an error in case `.sol` or `.eth` identity is passed for now.